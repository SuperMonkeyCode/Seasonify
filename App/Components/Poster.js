import React from 'react';
import FitImage from 'react-native-fit-image';

const Poster = ({ url }) => (
  <FitImage
    source={{ uri: url }}
    resizeMode="contain"
    resizeMethod="scale"
    indicator
    indicatorColor="#919191"
    indicatorSize="large"
  />
);

export default Poster;
