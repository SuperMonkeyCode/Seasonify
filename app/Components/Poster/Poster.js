/* @flow */
import { Image } from 'react-native';
import { compose, pure } from 'recompose';
import React from 'react';

import withResponsive from 'HOC/withResponsive';
import Metrics from 'Themes/Metrics';

type Props = {
  /* parent */
  url: string,
  orientation: 'PORTRAIT' | 'LANDSCAPE',
  window: Object,
  /* connect */
  /* HOC */
};

const enhance = compose(pure, withResponsive);

function Poster({ url, orientation, window }: Props) {
  const imageWidth = window.width * Metrics.columnLeft / 100;

  return (
    <Image
      source={{ uri: url }}
      resizeMode="cover"
      resizeMethod="scale"
      orientation={orientation}
      style={{
        height: imageWidth * Metrics.poster.averageRatio,
        width: imageWidth,
      }}
    />
  );
}

export default enhance(Poster);
