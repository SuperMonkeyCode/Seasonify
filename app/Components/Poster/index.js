/* @flow */
export { default } from './Poster';
