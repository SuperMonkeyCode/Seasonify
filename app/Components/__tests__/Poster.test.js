import React from 'react';
import Poster from '../Poster';

function setup(specificProps = {}) {
  const props = {
    url: 'http://somewhere.net/poster.jpg',
    ...specificProps,
  };
  const component = shallow(<Poster {...props} />);
  return {
    component,
    props,
  };
}

describe('Rendering', () => {
  it('should render', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
