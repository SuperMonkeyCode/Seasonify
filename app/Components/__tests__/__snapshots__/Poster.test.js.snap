// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering should render 1`] = `
<Image
  resizeMethod="scale"
  resizeMode="contain"
  source={
    Object {
      "uri": "http://somewhere.net/poster.jpg",
    }
  }
/>
`;
